src/binary-search-tree.js

```diff
@@ -4,6 +4,53 @@ class BinarySearchTree {
     this.value = value;
     this.left = null;
     this.right = null;
+    this.level = 0;
+  }
+  insert(value) {
+    const node = new BinarySearchTree(value);
+    const rec = (current) => {
+      node.level++;
+      if (current.value > value && !current.left) {
+        current.left = node;
+      } else if (current.value > value && current.left) {
+        rec(current.left);
+      } else if (current.value < value && !current.right) {
+        current.right = node;
+      } else if (current.value < value && current.right) {
+        rec(current.right);
+      }
+    };
+    rec(this);
+  }
+  contains(value) {
+    let bool = false;
+    const rec = (current) => {
+      if (current.value === value) bool = true;
```

Comment:
Since you aren't immediately returning here after finding the value you're looking for, your recursive function will continue to execute even after you've already found the value, which is extra work you don't need to be doing. Just have the function return when it has found the target.